CodeStyleExercise/ColorsViewController.swift

```diff
@@ -8,88 +8,70 @@
 
 import UIKit
 
-class ColorsViewController : UIViewController, UICollectionViewDataSource, UITextFieldDelegate
-{
-
-    class constants {
-        static let DFLTNUMCOLORS:Int=12
-    }
-
-    @IBOutlet weak var collectionView:UICollectionView!
-    @IBOutlet weak var colorCountTextField: UITextField!
-
-    var ct: Int? = 12
-
+final class ColorsViewController: UIViewController, UICollectionViewDataSource, UITextFieldDelegate {
+    
+    @IBOutlet private weak var collectionView: UICollectionView!
+    @IBOutlet private weak var colorCountTextField: UITextField!
+    
+    public var ct: Int? = 12
+    
     override func viewDidLoad() {
         super.viewDidLoad()
+        
         self.collectionView.register(UINib(nibName: "ColorsCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "Cell")
-
+        
         self.colorCountTextField.placeholder = "12"
     }
-
+    
     override func didReceiveMemoryWarning() {
         super.didReceiveMemoryWarning()
     }
-
-    // MARK: - Collection View Data Source Methods
-
-    func collectionView(_ collectionView:UICollectionView, numberOfItemsInSection section : Int) -> Int{
-        return ct!
+    
+    // MARK: - UICollectionViewDataSource
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return ct ?? 0
     }
-
+    
     func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-        var c = collectionView.dequeueReusableCell(withReuseIdentifier: "Cell", for: indexPath) as! ColorsCollectionViewCell
-//        let color = UIColor.random
-//        c.backgroundColor = color
-//        c.hexColorLbl.text = color.description
-        c.configureCellWithColor(color: random_ui_color())
-        return c
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "Cell", for: indexPath) as! ColorsCollectionViewCell
+        cell.configureCellWithColor(color: random_ui_color())
+        
+        return cell
     }
-
-
-
-
-    //////////////////////////////////
-    // MARK: - UITextFieldDelegate ///
-    //////////////////////////////////
-
+    
+    // MARK: - UITextFieldDelegate
+    
     func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
-        if string.characters.count>0{
-            if (textField.text! as NSString).replacingCharacters(in: range, with: string).characters.count <= 2 {
-                var str = string.components(separatedBy: NSCharacterSet.decimalDigits.inverted).joined() // Filters string by removing non-decimal digits
-                return string == str
-            } else {
-                return false
-            }
+        guard string.characters.count > 0 else { return true }
```

Comment:
Good use of the `guard` statement here! Take a look at the solution code for another example too.